### ISSUE TITLE
adplug: Playing some LDS file is very loud compared to recent adplay

### ISSUE THREAD

**Issue by jcelerier** [DeaDBeeF-Player/deadbeef#1678]
Given the song "Rock Garden" from the game tyrian : ([song available here](http://members.iinet.net.au/~vannevar/tyrian/tyrianlds.zip)) 

The output from DeaDBeeF is heavily distored during the choruses (also some other songs have clicks and pops). I disabled all the DSP options, replay gain, etc... and lowered the volume, to no avail.

When trying the same file with adplug's player (https://github.com/adplug/adplay-unix) the song seems pretty faithful to the original game however.

**Comment by Alexey-Yakovenko** [comment]
I can't seem to hear any distortion on the Rock Garden song (although I don't know how it's supposed to sound). It has certain parts which sound like electric guitars played with distortion.. but that doesn't seem to be what you're hearing. I don't hear any clicks and pops either.

**Comment by Alexey-Yakovenko** [comment]
Tried on 2 Mac and 1 Linux computers with ddb 0.7.2 and trunk, no issues found.

**Comment by Alexey-Yakovenko** [comment]
I went as far as installing adplay on linux, to compare the sound. Couldn't notice any difference.

**Comment by jcelerier** [comment]
well that's weird, thanks for trying... here's what I am hearing (first the recording of adplay, then the recording of deadbeef).
[out.flac.zip](https://github.com/Alexey-Yakovenko/deadbeef/files/637627/out.flac.zip)

**Comment by Alexey-Yakovenko** [comment]
There is difference. Not that I could call that a heavy distortion, but some parts of the sound are too loud in deadbeef. I can't see any code changing any internal adplug volume..

My guess is that adplug got some new OPL emulator, which sounds differently (the one used in deadbeef is pretty old).

**Comment by Alexey-Yakovenko** [comment]
I reopened, and slightly changed the bug title. We'll look into it when I have some time.

**Comment by jcelerier** [comment]
thanks :)

**Comment by Alexey-Yakovenko** [comment]
I updated adplug to latest, and this issue is resolved. It won't be included in 1.8.3 because of high risk -- I had to change how C++ support works in deadbeef, which may cause compatibility issues.
Local builds from master branch already work fine, CI builds are in progress.

There's still some remaining work to reapply certain adplug patches, which fix bugs and performance issues. Fortunately, adplug devs fixed most of the issues that I found over the years! So I don't have to re-patch everything.
